examples/custom_operator.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python3
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import strawberryfields as sf
+from strawberryfields.ops import *
+from strawberryfields.utils import operator
+
+
+@operator(ns=4)
+def prepare_op(q):
+    """
+    :param q: This operator is applied to 4 registers
+    """
+    S = Sgate(-1)
+    S | q[0]
+    S | q[1]
+    S | q[2]
+    S | q[3]
+
+
+@operator(ns=3)
+def circuit_op(v1, v2, q):
```

Comment:
Nice that it works with parameters! Does the operator decorator always require the qumode register `q` as the last argument?

Reply:
It works with parameters and also for class methods, like the example in the unit test: https://github.com/XanaduAI/strawberryfields/pull/36/files#diff-c487e7ec5c066aeaecbe95c769fdccaaR31

Yes, the last argument are always the register(s).